sdk/tables/data-tables/src/odata.ts

```diff
@@ -7,7 +7,7 @@ function escapeQuotesIfString(input: unknown, previous: string): string | unknow
   if (typeof input === "string") {
     result = escapeQuotes(input);
     // check if we need to escape this literal
-    if (!previous.trim().endsWith("'")) {
+    if (previous !== "" && !previous.trim().endsWith("'")) {
```

Comment:
I think we should have more unit tests for `odata`, as this has the potential to break more things than it fixes.

Reply:
Added a test for escaping strings!